plugins.go

```diff
@@ -18,7 +20,8 @@ func globalPluginDirs() []string {
 	if err != nil {
 		log.Printf("[ERROR] Error finding global config directory: %s", err)
 	} else {
-		ret = append(ret, filepath.Join(dir, "plugins"))
+		machineDir := fmt.Sprintf("%s_%s", runtime.GOOS, runtime.GOARCH)
+		ret = append(ret, filepath.Join(dir, "plugins", machineDir))
```

Comment:
yeah, I can see keeping the `~/.terraform.d/plugins` directory just to reduce churn in a point release. 

As for for machineDir, it seems there's a const in the command package that we're not using, and everywhere else is generating longer paths I can clean that up in another PR.